tests/pr_changed_files/test_all_match_patterns.sh

```diff
@@ -26,34 +26,52 @@ git add .
 git commit -m "Add doc files"
 
 # [Test] All changes in docs
-result=$(pr_changed_files --all-match "docs/*")
-assert_output "true" "$result" "Should return true when all changes match patterns"
+pr_changed_files --all-match "docs/*"
+result=$?
+assert_return_code 0 $result "Should return 0 when all changes match patterns"
```

Comment:
We probably still want to assert that `pr_changed_files` also echos true/false, in addition to testing the return code…

```suggestion
result=$(pr_changed_files --all-match "docs/*")
assert_return_code 0 $? "Should return 0 when all changes match patterns"
assert_output "true" "$result" "Should print true when all changes match patterns"
```

Though at that point it'll probably be worth it to create a combined assertion helper instead of 2 separate ones (given you'll always want to assert both in all test cases, I think), something like this maybe?
```
assert_result 0 "$?" "$result" "when all changes match patterns"
```

Reply:
> it'll probably be worth it to create a combined assertion helper instead of 2 separate ones 

Great idea 👍 implemented on b32c6b90529c09fdb7caf3a7877d315dcf328a55.